### PR TITLE
Add support for a netcore50 target

### PR DESCRIPTION
**Pull request by rynowak**
/cc @bricelam @anpete

### CODE REVIEW

File: src/Microsoft.Framework.Notification/Internal/ProxyAssembly.cs [hunk 1]
```diff
@@ -36,15 +34,6 @@ static ProxyAssembly()
...
         }
-
-#if NET45 || DNX451
-
-        public static void WriteToFile(string file)
-        {
-            AssemblyBuilder.Save(file);
-        }
-
-#endif
```
Comment (rynowak) [inline src/Microsoft.Framework.Notification/Internal/ProxyAssembly.cs]: This was old testing code that isn't needed.

File: src/Microsoft.Framework.Notification/Internal/ProxyAssembly.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+#if PROXY_SUPPORT
```
Comment (anpete) [inline src/Microsoft.Framework.Notification/Internal/ProxyAssembly.cs]: REF_EMIT?

File: src/Microsoft.Framework.Notification/ReflectionNotifierMethodAdapter.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+// We only need ReflectionNotifierMethodAdapter for netcore50 or any case where we can't use Reflection.Emit.
+//
+// However to test it, we compile it into each test assembly. This lets us write tests without a separate
+// flavor of tests for netcore50.
```
Comment (rynowak) [inline src/Microsoft.Framework.Notification/ReflectionNotifierMethodAdapter.cs]: This seemed like the best approach to make sure that we're testing this. We don't have any existing infrastructure to test `netcore50` directly.

File: test/Microsoft.Framework.Notification.Test/NotifierTest.cs [hunk 4]
```diff
@@ -200,7 +200,7 @@ public SomeValueType(int value)
 
         private static INotifier CreateNotifier()
         {
-            return new Notifier(new NotifierMethodAdapter());
+            return new Notifier(new ProxyNotifierMethodAdapter());
```
Comment (rynowak) [inline test/Microsoft.Framework.Notification.Test/NotifierTest.cs]: Just a rename since we have 2 implementations now.

**Comment by anpete** [comment]
:shipit:

**Pull request merged**